app/src/main/java/org/wikipedia/WikipediaApp.kt

```diff
@@ -79,13 +78,7 @@ class WikipediaApp : Application() {
         }
 
     val appOrSystemLanguageCode: String
-        get() {
```

Comment:
This getter is accessed from a lot of places which need to know the "default" app language code.
But notice that it also checks the User ID of the current user on the current language wiki, and if the User ID is 0, it _kicks off a network request_ to fetch the User ID. This is very inefficient, and is doing way too much in a hidden way.